trace_decoder/src/lib.rs

```diff
@@ -1,92 +1,63 @@
-//! <div class="warning">
-//! This library is undergoing major refactoring as part of (#275)(https://github.com/0xPolygonZero/zk_evm/issues/275).
-//! Consider all TODOs to be tracked under that issue.
-//! </div>
+//! An _Ethereum Node_[^1] executes _transactions_ in _blocks_.
 //!
-//! Your neighborhood zk-ready [ethereum](https://github.com/0xPolygonZero/erigon)
-//! [node](https://github.com/0xPolygonHermez/cdk-erigon/) emits binary "witnesses"[^1].
+//! Execution mutates two key data structures:
+//! - [The state trie](https://ethereum.org/en/developers/docs/data-structures-and-encoding/patricia-merkle-trie/#state-trie).
+//! - [The storage tries](https://ethereum.org/en/developers/docs/data-structures-and-encoding/patricia-merkle-trie/#storage-trie).
 //!
-//! But [`plonky2`], your prover, wants [`GenerationInputs`].
+//! Ethereum nodes expose information about the transactions over RPC, e.g:
+//! - [The specific changes to the storage tries](TxnTrace::storage_written).
+//! - [Changes to account balance in the state trie](TxnTrace::balance).
 //!
-//! This library helps you get there.
+//! The tries and the transactions can be proved to be correct using a
+//! zero-knowledge prover like [`evm_arithmetization`],
+//! an Ethereum-y frontend for [`plonky2`].
 //!
-//! [^1]: A witness is an attestation of the state of the world, which can be
-//!       proven by a prover.
+//! **Prover perfomance is a high priority.**
 //!
-//! # Non-Goals
-//! - Performance - this won't be the bottleneck in any proving system.
-//! - Robustness - malicious or malformed input may crash this library.
+//! The aformentioned trie structures may have subtries _deferred_.
+//! That is, any node (and its children!) may be replaced by its hash,
+//! while maintaining provability of its contents:
+//! ```text
+//!     A               A
+//!    / \             / \
+//!   B   C     ->    H   C
+//!  / \   \               \
+//! D   E   F               F
+//! ```
+//! (where `H` is the hash of the `D/B\E` subtrie).
 //!
-//! TODO(0xaatif): https://github.com/0xPolygonZero/zk_evm/issues/275
-//!                refactor all the docs below
+//! The principle concern of this library is to step through the transactions,
+//! and reproduce the tries that the Ethereum node does,
```

Comment:
well, _technically_, I would assume clients update the world state trie in a batch once processing txns, in which case we do _not_ mimic Ethereum here, but rather provide the _intermediate_ tries necessary for each batch of txn to be executed independently from each other